### PR TITLE
Rsarwad pthread condition variable

### PR DESCRIPTION
**Pull request by rsarwad**
[AGW] Modified code to wait until controller connects to switch

## Summary
On mme restart if Modify bearer Request arrives before controller connected to switch, it was making mme to crash. Because on reception of MBR message SPGW will invoke API to add OVS rules to Switch. Since connection is not established between controller and switch. It was resulting into crash.

Test plan:
Executed S1ap tester sanity test suite

### CODE REVIEW

**Comment by ssanadhya** [comment]
RestartIntegrationTests

**Comment by ssanadhya** [comment]
RestartLteTests

**Comment by ssanadhya** [comment]
RestartLTETests

**Comment by rsarwad** [comment]
> RestartLTETests

Hi @ssanadhya, How do I restartLteTests? I am not able to view the Jenkins report it generates only log files

File: lte/gateway/c/oai/lib/openflow/controller/ControllerMain.cpp [hunk 1]
```diff
@@ -15,6 +15,10 @@
  *      contact@openairinterface.org
  */
 
+#include <thread>
```
Comment (ssanadhya) [inline lte/gateway/c/oai/lib/openflow/controller/ControllerMain.cpp]: These headers are not being used here.

File: lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp [hunk 2]
```diff
@@ -15,14 +15,23 @@
  *      contact@openairinterface.org
  */
 
+#include <thread>
+#include <mutex>   // std::timed_mutex
```
Comment (ssanadhya) [inline lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp]: Remove comments

File: lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp [hunk 3]
```diff
@@ -15,14 +15,23 @@
...
+#include <chrono>  // std::chrono::seconds
```
Comment (ssanadhya) [inline lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp]: Remove comments

File: lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp [hunk 4]
```diff
@@ -94,12 +108,38 @@ void OpenflowController::dispatch_event(const ControllerEvent& ev) {
...
+    if (is_controller_connected_to_switch(CONNECTION_EVENT_WAIT_TIME) ==
+        RETURNok) {
+      OAILOG_INFO(LOG_GTPV1U, "Controller is now connected to switch \n");
+    }
   }
   ev->set_of_connection(latest_ofconn_);
   latest_ofconn_->add_immediate_event(cb, ev);
 }
 
+bool OpenflowController::is_controller_connected_to_switch(int duration) {
```
Comment (ssanadhya) [inline lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp]:
```suggestion
bool OpenflowController::is_controller_connected_to_switch(int conn_timeout) {
```

File: lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp [hunk 5]
```diff
@@ -94,12 +108,38 @@ void OpenflowController::dispatch_event(const ControllerEvent& ev) {
...
+  /* c++ provided conditional variable is added along with wait for seconds as
```
Comment (ssanadhya) [inline lte/gateway/c/oai/lib/openflow/controller/OpenflowController.cpp]: "Wait for conn_timeout seconds to make sure connection is established....."

File: lte/gateway/c/oai/lib/openflow/controller/OpenflowController.h [hunk 6]
```diff
@@ -115,6 +115,7 @@ class OpenflowController : public fluid_base::OFServer {
    */
   void inject_external_event(
       std::shared_ptr<ExternalEvent> ev, void* (*cb)(std::shared_ptr<void>) );
+  bool is_controller_connected_to_switch(int);
```
Comment (ssanadhya) [inline lte/gateway/c/oai/lib/openflow/controller/OpenflowController.h]: Please add parameter name here

**Review by ssanadhya — CHANGES_REQUESTED** [review]

**Review by ssanadhya — APPROVED** [review]

**Pull request merged**